examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1467 @@
+  {
+    TimerOutput::Scope t(computing_timer, "store_frequency_step_data");
+
+    // We store the displacement in the $x$ direction; the displacement in the
+    // $y$ direction is negligible.
+    const int probe_displacement_component = 0;
+
+    // The vector coordinates contains the coordinates in the HDF5 file of the
+    // points of the probe that are located in locally owned cells. The vector
+    // displacement_data contains the value of the displacement at these points.
+    std::vector<hsize_t>              coordinates;
+    std::vector<std::complex<double>> displacement_data;
+    for (unsigned int position_idx = 0;
+         position_idx < parameters.nb_probe_points;
+         ++position_idx)
+      {
+        Point<dim> point;
+        for (unsigned int dim_idx = 0; dim_idx < dim; ++dim_idx)
+          {
+            point(dim_idx) = position[position_idx][dim_idx];
+          }
+        bool point_in_locally_owned_cell;
+        {
+          // First we have to find out if the point is in a locally owned cell.
+          auto mapping = StaticMappingQ1<dim>::mapping;
+          const std::pair<typename DoFHandler<dim>::active_cell_iterator,
+                          Point<dim>>
+            cell_point = GridTools::find_active_cell_around_point(mapping,
+                                                                  dof_handler,
+                                                                  point);
+
+          point_in_locally_owned_cell = cell_point.first->is_locally_owned();
+        }
+        if (point_in_locally_owned_cell)
+          {
+            // Then we can store the values of the displacement in the points of
+            // the probe in `displacement_data`.
+            Vector<std::complex<double>> tmp_vector(dim);
+            VectorTools::point_value(dof_handler,
+                                     locally_relevant_solution,
+                                     point,
+                                     tmp_vector);
+            coordinates.emplace_back(position_idx);
+            coordinates.emplace_back(frequency_idx);
+            displacement_data.emplace_back(
+              tmp_vector(probe_displacement_component));
+          }
+      }
+
+    // We write the displacement data in the HDF5 file. The call
+    // HDF5::DataSet::write_selection() is MPI collective which means that all
+    // the processes have to participate.
+    if (coordinates.size() > 0)
+      {
+        displacement.write_selection(displacement_data, coordinates);
+      }
+    // Therefore even if the process has no data to write it has to participate
+    // in the collective call. For this we can use HDF5::DataSet::write_none().
+    // Note that we have to specify the data type, in this case
+    // `std::complex<double>`.
+    else
+      {
+        displacement.write_none<std::complex<double>>();
+      }
+
+    // If the variable of the jupyter notbook `save_vtu_files = True` then all
```

Comment:
Let's drop the reference to how the parameter file was created
```suggestion
    // If the variable `save_vtu_files` in the input file equals `True` then all
```